From 3a03aef2def46fee84fde38280ed176cbe63b761 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Fri, 8 Sep 2017 04:35:48 +0200
Subject: [PATCH] packages: virtualbox and pacman flags workarounds

- when vbox is detected install vbox guest package
- added -Scc to clean fpms
- added --noarch , --noprogressbar to flags
---
 src/modules/packages/main.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/modules/packages/main.py b/src/modules/packages/main.py
index 48caae6bd..c4a5ad999 100644
--- a/src/modules/packages/main.py
+++ b/src/modules/packages/main.py
@@ -240,7 +240,12 @@ class PMPacman(PackageManager):
             pacman_flags = "-Sy"
 
         check_target_env_call(["pacman", pacman_flags,
-                               "--noconfirm"] + pkgs)
+                               "--noconfirm", "--noprogressbar", "--noarch"] + pkgs)
+        vbox_sysfile = open("/sys/devices/virtual/dmi/id/bios_version", "r")
+        vbox_out = vbox_sysfile.read(10)
+        if 'VirtualBox' in vbox_out:
+            check_target_env_call(["pacman", pacman_flags, "--noconfirm", "virtualbox-guest-additions"])
+        check_target_env_call(["pacman", "-Scc", "--noconfirm"])
 
     def remove(self, pkgs):
         check_target_env_call(["pacman", "-Rs", "--noconfirm"] + pkgs)
-- 
2.14.1

